src/lambda/commands/deploySamApplication.ts

```diff
@@ -30,40 +31,71 @@ export async function deploySamApplication({
     }
 
     const { template, s3Bucket, stackName } = args
+    const deployApplicationPromise = new Promise<void>(async (resolve, reject) => {
 
-    const tempFolder = await mkdtemp('samDeploy')
-    const outputTemplatePath = path.join(tempFolder, 'template.yaml')
-    let stage = 'packaging'
-    try {
-        const packageInvocation = new SamCliPackageInvocation(template.fsPath, outputTemplatePath, s3Bucket, invoker)
-        restParams.outputChannel.show(true)
-        // TODO: Add nls support
-        restParams.outputChannel.appendLine(`Packaging SAM Application to S3 Bucket: ${s3Bucket}`)
-        await packageInvocation.execute()
+        const tempFolder = await mkdtemp('samDeploy')
+        const buildDestination = path.join(tempFolder, 'build')
+        const buildTemplatePath = path.join(buildDestination, 'template.yaml')
+        const outputTemplatePath = path.join(tempFolder, 'template.yaml')
+        let stage = 'starting up'
```

Comment:
Nit: Suggest to make `stage` type `'starting up' | 'building' | 'packaging'` to protect against typos.